docs/DeveloperGuide.adoc

```diff
@@ -572,6 +572,33 @@ Also creates coupling between the `Autocompleter` and `CommandBox`. +
 **Pros:** Easier to test since `CommandBoxTest` has already been set up and implemented. +
 **Cons:** `CommandBox` class now has multiple responsibilities, which violates SRP. +
 
+=== Maintain sorted
+The list of parcels in Ark is maintained to be always in sorted order according to delivery dates,
+with the earliest being on the top. This is so that the user will be able to look at the more
+pertinent deliveries. +
+
+The list is sorted whenever a parcel is added, edited or if a redo command is made. This is
+because these commands are the ones that might possibly cause the new parcel to be placed
+in the wrong position. +
+
+To maintain the selected element despite sorting however, we require a way to track which parcel
+ws selected. Static variables to track which card has been selected has been added to the Model
+class. The Model class now has the following methods: `select`, `unselect`, `getPrevIndex`,
+`hasSelected` and `forceSelect`. Whenever we select an item, we will use select to indicate that
+we have selected a parcel card and we will then re-select that card after we have sorted. +
+
+==== Design Considerations
+
+**Aspect:** Implementation of `maintainSorted` +
+**Alternative 1 (current choice):** Constant sort the list of parcels whenever there is a change that
+potentially could disrupt the order of the list. +
+**Pros:** Intuitive and guarantees that list is sorted in the right order +
+**Cons:** Many commands have to be changed +
+**Alternative 2:** Insert the new / edited parcel to fit into the sorted list. +
+**Pros:** Use less computation as the list of parcels is already sorted. +
+**Cons:** More difficult to implement as we'll need to implement our own sorting algorithm as opposed to
+just using the built in sorting methods. +
+
```

Comment:
Are there more design considerations you could add?

Reply:
Not really, but I'll try to think of more.